### ISSUE TITLE
TF version not learning correctly

### ISSUE THREAD

**Issue by mkroutikov** [mkroutikov/tf-lstm-char-cnn#1]
![image](https://cloud.githubusercontent.com/assets/14280777/18018137/35e72490-6ba3-11e6-9dd3-fc2f16ff92de.png)

Here blue and orange are different runs of TF training. Gray line is the original Torch7 code.

Clearly, TF version has bugs and does not learn correctly :(

**Comment by mkroutikov** [comment]
Okie, problem is the different gradient scaling in TF (comparing to Torch code).

In the Torch original code, gradients are accumulated for every iteration of time-unroll loop (i.e. 35 times for default value of seq_length=35), see [how gradients are zeroed out outside of time loop](https://github.com/yoonkim/lstm-char-cnn/blob/master/main.lua#L293)

TF rather computes gradients directly from the loss, and since loss is averaged over batch_size and num_unroll_steps, they are 35 time smaller.

This is not a problem per se, since one can use learning_rate=35 and max_grad_norm=0.15 to compensate for this. But in order to use the original values of parameters gradients has to be scaled up.

This commit fixes the issue: https://github.com/mkroutikov/tf-lstm-char-cnn/commit/fe0307a3113ae5bd9ee5520e029a8a80a247add2

After the fix, TF version seem to learn correctly:

![image](https://cloud.githubusercontent.com/assets/14280777/18178383/169b458c-704c-11e6-8b7c-428c7a92d41e.png)
